sklearn/utils/estimator_checks.py

```diff
@@ -2650,63 +2698,69 @@ def check_supervised_y_2d(name, estimator_orig):
 
 @ignore_warnings
 def check_classifiers_predictions(X, y, name, classifier_orig):
-    classes = np.unique(y)
-    classifier = clone(classifier_orig)
-    if name == "BernoulliNB":
-        X = X > X.mean()
-    set_random_state(classifier)
+    if name != "OneClassSampleErrorClassifierPredict":
```

Comment:
I do not think our common test should special case a custom testing estimator and skip it. Can `OneClassSampleErrorClassifierPredict` be created to pass `check_classifiers_predictions` but fail `check_classifiers_one_label_sample_weights`? If not, I prefer to have `OneClassSampleErrorClassifierPredict` xfail `check_classifiers_predictions` using estimator tags.

Reply:
I wasn't happy with this solution neither. Refactoring `OneClassSampleErrorClassifier` as [you suggested](https://github.com/scikit-learn/scikit-learn/pull/24140/files#r963090186) also solve this problem.
I have removed the exception then.